pkg/controller/gitopscluster/gitopscluster_controller.go

```diff
@@ -714,47 +720,89 @@ func (r *ReconcileGitOpsCluster) AddManagedClustersToArgo(
 }
 
 // CreateManagedClusterSecretInArgo creates a managed cluster secret with specific metadata in Argo namespace
-func (r *ReconcileGitOpsCluster) CreateManagedClusterSecretInArgo(argoNamespace string, managedClusterSecret v1.Secret) error {
-	labels := managedClusterSecret.GetLabels()
-
+func (r *ReconcileGitOpsCluster) CreateManagedClusterSecretInArgo(argoNamespace string, managedClusterSecret v1.Secret,
+	managedCluster string, enablePullModel bool) error {
 	// create the new cluster secret in the argocd server namespace
-	newSecret := &v1.Secret{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Secret",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      managedClusterSecret.Name,
-			Namespace: argoNamespace,
-			Labels: map[string]string{
-				"argocd.argoproj.io/secret-type":                 "cluster",
-				"apps.open-cluster-management.io/acm-cluster":    "true",
-				"apps.open-cluster-management.io/cluster-name":   labels["apps.open-cluster-management.io/cluster-name"],
-				"apps.open-cluster-management.io/cluster-server": labels["apps.open-cluster-management.io/cluster-server"],
+	var newSecret *v1.Secret
+
+	if enablePullModel {
+		newSecret = &v1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Secret",
+				APIVersion: "v1",
 			},
-		},
-		Type: "Opaque",
-		StringData: map[string]string{
-			"config": string(managedClusterSecret.Data["config"]),
-			"name":   string(managedClusterSecret.Data["name"]),
-			"server": string(managedClusterSecret.Data["server"]),
-		},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      managedCluster + "-cluster-secret",
+				Namespace: argoNamespace,
+				Labels: map[string]string{
+					"argocd.argoproj.io/secret-type":                 "cluster",
```

Comment:
When enablePullModel is on, we don't need to set up the label `argocd.argoproj.io/secret-type`. Or the dummy cluster  will be shown in the Argocd application UI  destination cluster dropdown list.

Reply:
ApplicationSet controller will not be able to generate Application for these pull model managed clusters if there are no secrets with this label.

I think this is a known issue and we will have to wrap up all our pull model work and present to the ArgoCD community and ask how to tackle some of these issues.

Reply:
agree